torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -2526,3 +2534,137 @@ def maybe_positional_arg_names(func):
             else:
                 result.append(name)
     return result
+
+
+def canonicalize(gmod, root_gmod):
+    # autograd_cache_key is sensitive to the name of the placeholder nodes.
```

Comment:
Are we sure it's not sensitive to intermediate node names too? Can we add a test for this?

Reply:
We are. The comment is outdated. Let me change and add a test.